### PR TITLE
Use weights_only for load

### PR DESCRIPTION
**Pull request by kit1980**
`torch.load` without `weights_only` parameter is unsafe. Explicitly set `weights_only` to False only if you trust the data you load and full pickle functionality is needed, otherwise set `weights_only=True`.

If `weights_only=True` doesn't work for some cases, then explicit `weights_only=False` should be used.

Found with https://github.com/pytorch-labs/torchfix/

### CODE REVIEW

File: azula/plugins/adm/__init__.py [hunk 1]
```diff
@@ -187,7 +187,7 @@ def load_model(key: str, **kwargs) -> ImprovedDenoiser:
         state = torch.load(
             f=cached_download(url=url),
             **kwargs,
-        )
+        weights_only=True)
```
Comment (francois-rozet) [inline azula/plugins/adm/__init__.py:190]: `weights_only=True` is already set by default in the `kwargs` dictionary.

File: tests/test_nn_unet.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def test_UNet(
 
     # Load
     copy = make()
-    copy.load_state_dict(torch.load(tmp_path / "state.pth"))
+    copy.load_state_dict(torch.load(tmp_path / "state.pth", weights_only=True))
```
Comment (francois-rozet) [inline tests/test_nn_unet.py:68]: :+1:

**Review by francois-rozet — COMMENTED** [review]
Thank you for the PR! This is the very first PR of Azula :partying_face:

I agree that `weights_only=True` should be used as much as possible. However, we already set it to true by default, and I am not sure it necessary to enforce it further.

**Comment by kit1980** [comment]
> Do you know the version it was introduced in?

The PR that added the option is https://github.com/pytorch/pytorch/pull/86812, first release with it is PyTorch 1.13.0

File: azula/plugins/adm/__init__.py (hunk 1 above)
Comment (kit1980) [reply azula/plugins/adm/__init__.py:190]: I see now. Then it's already fine, just hard to see with static analysis.

**Comment by francois-rozet** [comment]
Thank you! I have dropped the support for `torch<=1.12` to be able to use `weights_only` everywhere.

I also plan to add a hash-check protocol for our plugins to ensure that downloaded files are safe to read. I have already implemented part of that protocol in #6.

**Pull request merged**